src/test/scala/is/hail/linalg/BlockMatrixSuite.scala

```diff
@@ -1007,6 +1007,7 @@ class BlockMatrixSuite extends SparkSuite {
     
     TestUtils.interceptFatal(notSupported) { bm0.rowVectorDiv(v0) }
     TestUtils.interceptFatal(notSupported) { bm0.colVectorDiv(v0) }
+    TestUtils.interceptFatal("cannot multiply block matrix by scalar") { bm0 * Double.NaN }
```

Comment:
I'm moving the check to only the sparse case, since we shouldn't needlessly step on normal rules for doubles. I'd prefer to leave the sparse matrix restrictions in Scala as I increase what's supported.